contentcuration/contentcuration/frontend/channelEdit/components/sidePanels/SubmitToCommunityLibrarySidePanel/LicenseStatus.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+
+  <Box
+    :kind="boxKind"
+    data-test="license-status"
+  >
+    <template #title>
+      {{ titleText }}
+    </template>
+    <template #description>
+      {{ descriptionText }}
+    </template>
+  </Box>
+
+</template>
+
+
+<script>
+
+  import { computed } from 'vue';
+  import Box from './Box.vue';
+  import { formatLicenseNames } from './composables/useLicenseNames';
+  import { findLicense } from 'shared/utils/helpers';
+  import { communityChannelsStrings } from 'shared/strings/communityChannelsStrings';
+
+  const {
+    licenseCheckPassed$,
+    allLicensesCompatible$,
+    incompatibleLicensesDetected$,
+    channelCannotBeDistributed$,
+    fixLicensingBeforeSubmission$,
+  } = communityChannelsStrings;
+
+  export default {
+    name: 'LicenseStatus',
+    components: {
+      Box,
+    },
+    setup(props) {
+      const invalidLicenseNames = computed(() => formatLicenseNames(props.invalidLicenses));
+
+      const includedLicenseNames = computed(() => {
+        if (!props.includedLicenses) return '';
+
+        return props.includedLicenses
+          .map(id => {
+            const license = findLicense(id);
+            return license.license_name;
+          })
+          .filter(name => name !== 'Special Permissions')
+          .join(', ');
+      });
+
+      const hasInvalidLicenses = computed(() => {
+        return props.invalidLicenses && props.invalidLicenses.length > 0;
```

Comment:
A more direct way for null checking is using the optional chaining operator: `?.`

```javascript
props.invalidLicenses?.length > 0
```